### PR TITLE
Tcl test runner upgraded to spec version 2

### LINKED ISSUE #8: Upgrade to version 2 spec
If possible, this test runner should be updated to version 2 of the [test runner interface specification](https://github.com/exercism/docs/blob/main/building/tooling/test-runners/interface.md).

The version 2 spec differs from version 1 in that the `results.json` file contains detailed information on individual tests.
This information includes things like:

- The test status (passing/failing)
- The error message (if failing)
- The test code that was used to run the test

You are completely free on how you want to extract the test code, either via the AST or via finding the test code in the source code text via string manipulation. 

Let me know if there are any questions.

### PR DESCRIPTION
**Pull request by glennj**
Resolves #8

Of note, this PR changes
* the implementation language of `run.sh` to `run.tcl`
* the Docker FROM image to pull in a popular 3rd party JSON library for Tcl. Work well on my laptop. The image size is larger than the base ubuntu image previously used: was 123MB, is now 224MB. I trust that won't be an issue in the cloud infrastructure.

### CODE REVIEW

File: tests/example-all-fail/expected_results.json [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+    "version": 2,
+    "status":  "fail",
+    "message": null,
+    "tests":   [
+        {
+            "name":      "leap-1",
+            "status":    "fail",
+            "message":   "==== leap-1 year not divisible by 4 in common year FAILED\n==== Contents of test case:\n\n    isLeapYear 2015\n\n---- Result was:\n1\n---- Result should have been (boolean matching):\nfalse\n==== leap-1 FAILED",
+            "output":    null,
+            "test_code": "\n    isLeapYear 2015\n"
```
Comment (ErikSchierboom) [inline tests/example-all-fail/expected_results.json:11]: Maybe trim this?

```suggestion
            "test_code": "\n    isLeapYear 2015\n"
```

**Review by ErikSchierboom — APPROVED** [review]
Looks good! Just one tiny suggestion.

File: tests/example-all-fail/expected_results.json (hunk 1 above)
Comment (glennj) [reply tests/example-all-fail/expected_results.json:11]: Sorry, I don't see the difference. Do you mean trim leading and trailing whitespace?
```
            "test_code": "isLeapYear 2015"
```

Comment (ErikSchierboom) [reply tests/example-all-fail/expected_results.json:11]: Eh, ah yes, sorry about that.

Comment (glennj) [reply tests/example-all-fail/expected_results.json:11]: OK will do.

**Pull request merged**